tests/embedded-common-test.cpp

```diff
@@ -0,0 +1,117 @@
+#include "sensirion_test_setup.h"
+#include "sensirion_shdlc.h"
+#include "sensirion_i2c.h"
+
+TEST_GROUP (EmbeddedCommon_SHDLC_Tests) {
+    void setup() {
+    }
+
+    void teardown() {
+    }
+};
+
+TEST_GROUP (EmbeddedCommon_I2C_Tests) {
+    void setup() {
+    }
+
+    void teardown() {
+    }
+};
+
+TEST (EmbeddedCommon_SHDLC_Tests, Frame_Test_1) {
+    uint16_t i;
+    uint8_t outdata[] = {0x7E, 0x00, 0x80, 0x05, 0x00, 0x00, 0x00, 0x00, 0x00, 0x7A, 0x7E};
+    uint8_t buffer[11];
+    uint32_t auto_clean_interval = 0;
+    uint8_t command = 0x80;
+    uint8_t address = 0;
+    uint8_t sub_command = 0x00;
+    uint8_t data_length = 5; // sub_command(1) + auto_clean_interval(4)
+    struct sensirion_shdlc_buffer frame;
+    sensirion_shdlc_begin_frame(&frame, &buffer[0],command,address,data_length);
+    sensirion_shdlc_add_uint8_t_to_frame(&frame,sub_command);
+    sensirion_shdlc_add_uint32_t_to_frame(&frame,auto_clean_interval);
+    sensirion_shdlc_finish_frame(&frame);
+    printf("\nStart testing SHDLC frame 1!\n");
+    for(i = 0; i < 11; i++){
+        printf("Expected: %02X Actual: %02X\n",outdata[i],buffer[i]);
+        CHECK(buffer[i] == outdata[i]);
```

Comment:
I propose that you use `MEMCMP_EQUAL` and remove the loop. Alternatively use `BYTES_EQUAL` and remove the `printf` line. Using the `*_EQUAL` macros will print what each expression and value was when it fails.

Reply:
Generally I think most of the time test shouldn't output anything when they succeed.